dyn/resource_dyn_record_test.go

```diff
@@ -163,6 +163,54 @@ func TestAccDynRecord_CNAME_trailingDot(t *testing.T) {
 	})
 }
 
+func TestAccDynRecord_NS_record(t *testing.T) {
+	var record dynect.Record
+	zone := os.Getenv("DYN_ZONE")
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckDynRecordDestroy,
+		Steps: []resource.TestStep{
+			resource.TestStep{
+				Config: fmt.Sprintf(testAccCheckDynRecordConfig_NS_record, zone),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckDynRecordExists("dyn_record.foobar", &record),
+					resource.TestCheckResourceAttr("dyn_record.foobar", "name", "dev"),
+					resource.TestCheckResourceAttr("dyn_record.foobar", "type", "NS"),
+					resource.TestCheckResourceAttr("dyn_record.foobar", "ttl", "3600"),
+					resource.TestCheckResourceAttr("dyn_record.foobar", "zone", zone),
+					resource.TestCheckResourceAttr("dyn_record.foobar", "value", "ns.terraform.io."),
+				),
+			},
+		},
+	})
+}
+
+func TestAccDynRecord_MX_record(t *testing.T) {
+	var record dynect.Record
+	zone := os.Getenv("DYN_ZONE")
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckDynRecordDestroy,
+		Steps: []resource.TestStep{
+			resource.TestStep{
+				Config: fmt.Sprintf(testAccCheckDynRecordConfig_MX_record, zone),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckDynRecordExists("dyn_record.foobar", &record),
+					resource.TestCheckResourceAttr("dyn_record.foobar", "name", "mail-test"),
+					resource.TestCheckResourceAttr("dyn_record.foobar", "type", "MX"),
+					resource.TestCheckResourceAttr("dyn_record.foobar", "ttl", "30"),
+					resource.TestCheckResourceAttr("dyn_record.foobar", "zone", zone),
+					resource.TestCheckResourceAttr("dyn_record.foobar", "value", "10 mx.terraform.io."),
```

Comment:
same question, why isn't this using `hashicorptest.com`?

Reply:
I think it doesn't matter in this context - it was kind of intentional as many folks use 3rd party providers for emails, so MX records would likely in practice point outside of the zone, unless you run your own mail server, which not so many people do these days.